miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt

```diff
@@ -55,20 +60,16 @@ internal class MiniAppDownloader(
     ) = apiClient.fetchFileList(appId, versionId)
 
     @SuppressWarnings("LongMethod")
-    private suspend fun downloadMiniApp(
-        appId: String,
-        versionId: String,
-        manifest: ManifestEntity
-    ): String {
+    private suspend fun downloadMiniApp(miniAppInfo: MiniAppInfo, manifest: ManifestEntity): String {
+        val appId = miniAppInfo.id
+        val versionId = miniAppInfo.version.versionId
         val baseSavePath = storage.getMiniAppVersionPath(appId, versionId)
         when {
             isManifestValid(manifest) -> {
                 for (file in manifest.files) {
                     val response = apiClient.downloadFile(file)
                     storage.saveFile(file, baseSavePath, response.byteStream())
                 }
-                miniAppStatus.setVersionDownloaded(appId, versionId, true)
-                miniAppStatus.saveDownloadedVersion(appId, versionId)
```

Comment:
Just curious to know, could we call `storeDownloadedMiniApp(miniAppInfo)` here?

Reply:
There is the reason why I changed the call of this method in `getMiniApp` function.

We cached only `versionId` before and now replace it with `miniAppInfo`. Our SDK checks that there is cache but no `miniAppInfo` saved so we need to update the status of downloaded miniapp even it's already downloaded.